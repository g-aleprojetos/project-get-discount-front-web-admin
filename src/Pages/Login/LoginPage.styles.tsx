import styled from 'styled-components';
import cores from 'resources/colors';
import {TextH1, TextH2, TextH5} from 'Components/text';

export interface PropsTelaLoginECadastro
  extends React.AllHTMLAttributes<HTMLElement> {
  tela?: 'Login' | 'Password' | 'Singn Up';
  ativo?: boolean;
}

export const BotaoRecuperaSenha = styled.button`
  background-color: transparent;
  border: none;
  margin-top: 10px;
  margin-bottom: 10px;
  cursor: pointer;
`;

export const Form = styled.form`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-width: 400px;
  padding: 20px;
  gap: 8px;
  background-color: ${cores.white};
  border-color: ${cores.black};
  border-radius: 20px;
  box-shadow: 0px 30px 80px 3px black;
`;

export const Main = styled.main`
  width: 100vw;
  height: 100vh;
  position: fixed;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: ${cores.lightgray};
`;

export const TextoBotao = styled(TextH2).attrs({
  cursor: 'pointer',
})``;

export const TextoRecuperarSenha = styled(TextH5).attrs({
  cor: cores.black,
  cursor: 'pointer',
})``;

export const TextoTitulo = styled(TextH1).attrs({
  cor: cores.blue,
})``;
