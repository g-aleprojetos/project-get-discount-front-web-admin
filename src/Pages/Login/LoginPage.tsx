import React, {useState} from 'react';
import {FormLogin} from './formLogin';
import {FormRecoverPassword} from './formRecoverPassword';
import * as S from './LoginPage.styles';

export const LoginPage = () => {
  const [recuperarSenha, setRecuperarSenha] = useState<boolean>(false);

  const handleEsqueceuSenha = () => {
    setRecuperarSenha(prev => !prev);
  };

  return (
    <S.Main data-testid={'loginPage'}>
      <FormLogin handleEsqueceuSenha={handleEsqueceuSenha} />

      {recuperarSenha && (
        <FormRecoverPassword handleEsqueceuSenha={handleEsqueceuSenha} />
      )}
    </S.Main>
  );
};
