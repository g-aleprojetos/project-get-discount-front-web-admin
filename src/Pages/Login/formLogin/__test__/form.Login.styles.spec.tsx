import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from '../form.Login.styles';

describe('form.Login.styles', () => {
  test('BotaoEnviar DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.BotaoEnviar />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('ContainerBotaoRecuperaSenha DEVE ser igual ao snapshot', () => {
    const container = renderer
      .create(<S.ContainerBotaoRecuperaSenha />)
      .toJSON();

    expect(container).toMatchSnapshot();
  });

  test('ContainerRecuperarSenha DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.ContainerRecuperarSenha />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Form DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.Form />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('TextoBotao DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.TextoBotao />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('TextoRecuperarSenha DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.TextoRecuperarSenha />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('TextoTitulo DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.TextoTitulo />).toJSON();

    expect(container).toMatchSnapshot();
  });
});
