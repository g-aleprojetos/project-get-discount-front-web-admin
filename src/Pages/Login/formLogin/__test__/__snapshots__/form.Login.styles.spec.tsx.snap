// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form.Login.styles BotaoEnviar DEVE ser igual ao snapshot 1`] = `
.c0 {
  margin: 15px;
}

.c1 {
  min-width: 200px;
  background-color: #3D9DB3;
  border-radius: 8px;
  border: none;
  padding: 8px 60px;
  cursor: pointer;
}

<div
  className="c0"
  data-testid="bnt-pardao"
>
  <button
    className="c1"
    data-testid="bnt-pardao-button"
    type="submit"
  />
</div>
`;

exports[`form.Login.styles ContainerBotaoRecuperaSenha DEVE ser igual ao snapshot 1`] = `
.c0 {
  background-color: transparent;
  border: none;
  cursor: pointer;
}

<button
  className="c0"
/>
`;

exports[`form.Login.styles ContainerRecuperarSenha DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  width: 100%;
  padding-right: 8px;
}

<div
  className="c0"
/>
`;

exports[`form.Login.styles Form DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 400px;
  padding: 20px;
  gap: 8px;
  background-color: #FFFFFF;
  border-color: #000000;
  border-radius: 20px;
  box-shadow: 0px 30px 80px 3px black;
}

<form
  className="c0"
/>
`;

exports[`form.Login.styles TextoBotao DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 18px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: 26px;
  cursor: pointer;
}

<h2
  className="c0 "
  cursor="pointer"
/>
`;

exports[`form.Login.styles TextoRecuperarSenha DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 16px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: 24px;
  cursor: pointer;
}

<h3
  className="c0 "
  cursor="pointer"
/>
`;

exports[`form.Login.styles TextoTitulo DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 24px;
  font-family: AvertaStd-Regular;
  color: #3D9DB3;
  margin-top: 0px;
  text-transform: none;
  line-height: 34px;
  cursor: text;
}

<h1
  className="c0 "
/>
`;
